lib/lpm.c

```diff
@@ -60,7 +66,36 @@ lpm_lookup_ipv4(struct rte_lpm *lpm, uint32_t ip)
 		G_LOG(ERR, "lpm: incorrect arguments for IPv4 lookup\n");
 		goto out;
 	} else if (ret == -ENOENT) {
-		G_LOG(WARNING, "lpm: IPv4 lookup miss\n");
+		/*
+		 * Failing to find an LPM entry can mean many different
+		 * things, depending on the caller. In some cases,
+		 * failing to find an entry is used as a validation of
+		 * the table, and is therefore not even an error.
+		 *
+		 * Failing to find an entry can also occur frequently
+		 * on the hot path when under attack.
+		 *
+		 * For these reasons, we are careful about how we log the
+		 * fact that the lookup failed to find an entry. We use the
+		 * DEBUG level and guess whether the entry will actually be
+		 * logged. Callers are encouraged to check for -ENOENT and
+		 * make their own log entries as needed.
+		 */
+		char buf[INET_ADDRSTRLEN];
+
+		if (!G_LOG_CHECK(DEBUG)) {
+			G_LOG(DEBUG, "lpm: IPv4 lookup miss\n");
+			goto out;
+		}
+
+		if (inet_ntop(AF_INET, &ip, buf, sizeof(buf)) != NULL) {
+			G_LOG(DEBUG,
+				"lpm: IPv4 lookup miss for %s\n", buf);
+		} else {
+			G_LOG(DEBUG,
+				"lpm: IPv4 lookup miss; can't convert IP to string: %s\n",
+				strerror(errno));
+		}
 		goto out;
```

Comment:
The label `out` (I can't comment there) is just a `return ret;`. We can avoid the label/gotos and just return.

The same for the IPv6 case.